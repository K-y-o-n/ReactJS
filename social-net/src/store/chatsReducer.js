import { ADD_CHAT, DELETE_CHAT } from "./types/chatsTypes";

const InitialState = { chatList: [{ id: "id0", name: "ГЛаДОС" }] };
let count = 0; // для того что бы, при удалении чатов, id новых чатов оставался уникальным

function chatsReducer(state = InitialState, action) {
  switch (action.type) {
    case ADD_CHAT:
      count++;
      return {
        ...state,
        chatList: [
          ...state.chatList,
          {
            id: `id${count}`,
            name: action.payload,
          },
        ],
      };

    case DELETE_CHAT:
      return {
        ...state,
        chatList: state.chatList.filter((el) => el.id !== action.payload),
      };

    default:
      return state;
  }
}

export default chatsReducer;
