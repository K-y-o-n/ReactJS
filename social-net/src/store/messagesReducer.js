import { ADD_MESSAGE } from "./types/messagesTypes";

const initialState = {
  // будет храниться в виде {[chatId]: [{id, text, author}]}
  messageList: {
    ["id0"]: [
      {
        id: 0,
        text: "Добро пожаловать в компьютеризированный экспериментальный центр при лаборатории исследования природы ReactJS!",
        author: "ГЛаДОС",
      },
      {
        id: 1,
        text: "Перед началом тестирования хотим вам напомнить, что хотя основным принципом экспериментального центра является обучение в игровой форме, мы не гарантируем отсутствие увечий и травм.",
        author: "ГЛаДОС",
      },
    ],
  },
};

function messegesReducer(state = initialState, action) {
  switch (action.type) {
    case ADD_MESSAGE: {
      const currentList = state.messageList[action.chatId] || [];
      return {
        ...state,
        messageList: {
          ...state.messageList,
          [action.chatId]: [
            ...currentList,
            {
              id: currentList.length,
              text: action.message,
              author: action.author,
            },
          ],
        },
      };
    }
    default:
      return state;
  }
}

export default messegesReducer;
