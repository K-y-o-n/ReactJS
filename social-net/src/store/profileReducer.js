// const InitialState = {
//   id1: {
//     name: "ГЛаДОС",
//     messages: [
//       {
//         text: "Добро пожаловать в компьютеризированный экспериментальный центр при лаборатории исследования природы ReactJS!",
//         author: "ГЛаДОС",
//       },
//       {
//         text: "Перед началом тестирования хотим вам напомнить, что хотя основным принципом экспериментального центра является обучение в игровой форме, мы не гарантируем отсутствие увечий и травм.",
//         author: "ГЛаДОС",
//       },
//     ],
//   },
//   id2: {
//     name: "Test",
//     messages: [{ text: "Test Messege!", author: "Bot" }],
//   },
// };
import CHANGE_NAME from "./types/profileTypes";

const InitialState = { name: "Без имени" };

function profileReducer(state = InitialState, { type, payload }) {
  switch (type) {
    case CHANGE_NAME:
      return { name: payload };
    default:
      return state;
  }
}

export default profileReducer;
