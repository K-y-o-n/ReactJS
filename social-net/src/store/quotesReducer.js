import { STATUSES } from "../action/quotesAction";
import {
  GET_QUOTE_REQUEST,
  GET_QUOTE_SUCCESS,
  GET_QUOTE_FAILURE,
} from "./types/quotesTypes";

const InitialState = {
  quotesList: [],
  loading: false,
  request: STATUSES.IDLE,
  error: null,
};

function quotesReducer(state = InitialState, action) {
  switch (action.type) {
    case GET_QUOTE_REQUEST:
      return {
        ...state,
        request: STATUSES.REQUEST,
        loading: true,
        error: null,
      };

    case GET_QUOTE_SUCCESS:
      // как лучше добавлять цитаты в массив quotes : через ...state или через push?
      // state.quotes.push(action.payload);
      return {
        ...state,
        quotesList: [...state.quotesList, action.payload],
        request: STATUSES.SUCCESS,
        loading: false,
      };

    case GET_QUOTE_FAILURE:
      return {
        ...state,
        request: STATUSES.FAILURE,
        error: action.payload,
        loading: false,
      };

    default:
      return state;
  }
}

export default quotesReducer;
