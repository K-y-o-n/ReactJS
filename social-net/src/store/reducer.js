// const InitialState = {
//   id1: {
//     name: "ГЛаДОС",
//     messages: [
//       {
//         text: "Добро пожаловать в компьютеризированный экспериментальный центр при лаборатории исследования природы ReactJS!",
//         author: "ГЛаДОС",
//       },
//       {
//         text: "Перед началом тестирования хотим вам напомнить, что хотя основным принципом экспериментального центра является обучение в игровой форме, мы не гарантируем отсутствие увечий и травм.",
//         author: "ГЛаДОС",
//       },
//     ],
//   },
//   id2: {
//     name: "Test",
//     messages: [{ text: "Test Messege!", author: "Bot" }],
//   },
// };

const InitialState = { checkbox: false };

function reducer(state = InitialState, action) {
  switch (action.type) {
    case "SET_CHECKBOX":
      return { ...state, checkbox: !state.checkbox };
    default:
      return state;
  }
}

export default reducer;
