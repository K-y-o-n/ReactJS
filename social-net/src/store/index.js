import { createStore, combineReducers, applyMiddleware } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import profileReducer from "./profileReducer";
import chatsReducer from "./chatsReducer";
import messagesReducer from "./messagesReducer";
import quotesReducer from "./quotesReducer";
import thunk from "redux-thunk";
import { persistStore, persistReducer } from "redux-persist";
import storage from "redux-persist/lib/storage"; // localStorage

// создаем объект конфигурации для persist
const persistConfig = {
  key: "socialNet",
  storage,
  blacklist: ["quotes"],
};

const rootReducer = combineReducers({
  profile: profileReducer,
  chats: chatsReducer,
  messages: messagesReducer,
  quotes: quotesReducer,
});

// оборачиваем редьюсеры в persist
const persistedReducer = persistReducer(persistConfig, rootReducer);

// создаем store с использованием persistedReducer
export const store = createStore(
  persistedReducer,
  composeWithDevTools(applyMiddleware(thunk))
);

// создаем persistor
export const persistor = persistStore(store);
