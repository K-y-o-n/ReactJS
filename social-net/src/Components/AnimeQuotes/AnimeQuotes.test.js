import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { store } from "../../store";
import AnimeQuotes from "./AnimeQuotes";

it("Создание снепшота компонента AnimeQuotes", () => {
  const component = render(
    <Provider store={store}>
      <BrowserRouter>
        <AnimeQuotes />
      </BrowserRouter>
    </Provider>
  );
  expect(component).toMatchSnapshot();
});
