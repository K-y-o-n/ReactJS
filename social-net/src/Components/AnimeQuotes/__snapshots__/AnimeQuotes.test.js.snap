// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot with no article 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <ul
          class="header__links"
        >
          <li>
            <a
              href="/profile"
              style="text-decoration: none;"
            >
              Профиль
            </a>
          </li>
          <li>
            <a
              href="/chats"
              style="text-decoration: none;"
            >
              Чаты
            </a>
          </li>
          <li>
            <a
              href="/quotes"
              style="text-decoration: none;"
            >
              Аниме цитаты
            </a>
          </li>
        </ul>
        <div
          class="header__authBtns"
        >
          <button>
            Вход
          </button>
        </div>
      </header>
      <div
        class="quotes"
      >
        <p
          class="quotes__p"
        >
          Отправьте пустое сообщение что бы получить цитату из случайного аниме. Или введите название аниме на латинице для цитаты из конкретного аниме.
        </p>
        <ul
          class="quotes__ul"
        />
      </div>
      <form
        class="form"
      >
        <div
          class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="filled-basic-small"
            id="filled-basic-small-label"
          >
            Сообщение
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="filled-basic-small"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Отправить
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <ul
        class="header__links"
      >
        <li>
          <a
            href="/profile"
            style="text-decoration: none;"
          >
            Профиль
          </a>
        </li>
        <li>
          <a
            href="/chats"
            style="text-decoration: none;"
          >
            Чаты
          </a>
        </li>
        <li>
          <a
            href="/quotes"
            style="text-decoration: none;"
          >
            Аниме цитаты
          </a>
        </li>
      </ul>
      <div
        class="header__authBtns"
      >
        <button>
          Вход
        </button>
      </div>
    </header>
    <div
      class="quotes"
    >
      <p
        class="quotes__p"
      >
        Отправьте пустое сообщение что бы получить цитату из случайного аниме. Или введите название аниме на латинице для цитаты из конкретного аниме.
      </p>
      <ul
        class="quotes__ul"
      />
    </div>
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="filled-basic-small"
          id="filled-basic-small-label"
        >
          Сообщение
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            id="filled-basic-small"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Отправить
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Создание снепшота компонента AnimeQuotes 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <ul
          class="header__links"
        >
          <li>
            <a
              href="/profile"
              style="text-decoration: none;"
            >
              Профиль
            </a>
          </li>
          <li>
            <a
              href="/chats"
              style="text-decoration: none;"
            >
              Чаты
            </a>
          </li>
          <li>
            <a
              href="/quotes"
              style="text-decoration: none;"
            >
              Аниме цитаты
            </a>
          </li>
        </ul>
        <div
          class="header__authBtns"
        >
          <button>
            Вход
          </button>
        </div>
      </header>
      <div
        class="quotes"
      >
        <p
          class="quotes__p"
        >
          Отправьте пустое сообщение что бы получить цитату из случайного аниме. Или введите название аниме на латинице для цитаты из конкретного аниме.
        </p>
        <ul
          class="quotes__ul"
        />
      </div>
      <form
        class="form"
      >
        <div
          class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
        >
          <label
            class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for="filled-basic-small"
            id="filled-basic-small-label"
          >
            Сообщение
          </label>
          <div
            class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
              id="filled-basic-small"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          Отправить
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <ul
        class="header__links"
      >
        <li>
          <a
            href="/profile"
            style="text-decoration: none;"
          >
            Профиль
          </a>
        </li>
        <li>
          <a
            href="/chats"
            style="text-decoration: none;"
          >
            Чаты
          </a>
        </li>
        <li>
          <a
            href="/quotes"
            style="text-decoration: none;"
          >
            Аниме цитаты
          </a>
        </li>
      </ul>
      <div
        class="header__authBtns"
      >
        <button>
          Вход
        </button>
      </div>
    </header>
    <div
      class="quotes"
    >
      <p
        class="quotes__p"
      >
        Отправьте пустое сообщение что бы получить цитату из случайного аниме. Или введите название аниме на латинице для цитаты из конкретного аниме.
      </p>
      <ul
        class="quotes__ul"
      />
    </div>
    <form
      class="form"
    >
      <div
        class="MuiFormControl-root MuiTextField-root form__text css-p21bt9-MuiFormControl-root-MuiTextField-root"
      >
        <label
          class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-filled MuiFormLabel-root MuiFormLabel-colorPrimary Mui-focused css-bohyag-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink="true"
          for="filled-basic-small"
          id="filled-basic-small-label"
        >
          Сообщение
        </label>
        <div
          class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl MuiInputBase-sizeSmall css-cio0x1-MuiInputBase-root-MuiFilledInput-root"
        >
          <input
            aria-invalid="false"
            class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-c5v1qu-MuiInputBase-input-MuiFilledInput-input"
            id="filled-basic-small"
            type="text"
            value=""
          />
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root form__submit css-ke5b6m-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="submit"
      >
        Отправить
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
